lib/srv/desktop/discovery.go

```diff
@@ -66,16 +66,26 @@ func (s *WindowsService) startDesktopDiscovery(ctx context.Context) error {
 
 		GetCurrentResources: func() types.ResourcesWithLabels { return s.lastDiscoveryResults },
 		GetNewResources:     s.getDesktopsFromLDAP,
-		OnCreate:            s.createDesktop,
-		OnUpdate:            s.updateDesktop,
-		OnDelete:            s.deleteDesktop,
-		Log:                 s.cfg.Log,
+		// On the first run of the reconcile loop, a desktop may exist in the database
+		OnCreate: s.createDesktop,
+		OnUpdate: s.updateDesktop,
+		OnDelete: s.deleteDesktop,
+		Log:      s.cfg.Log,
 	})
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	go func() {
+		// initialize lastDiscoverResults with any desktops from a previous run that were saved in the backend
```

Comment:
```suggestion
		// initialize lastDiscoveryResults with any desktops that already exist
```

It may not be a previous run, it may be from another Windows Desktop Service